docs/content/development/extensions-core/kafka-ingestion.md

```diff
@@ -115,7 +115,8 @@ The tuningConfig is optional and default parameters will be used if no tuningCon
 |Field|Type|Description|Required|
 |-----|----|-----------|--------|
 |`type`|String|The indexing task type, this should always be `kafka`.|yes|
-|`maxRowsInMemory`|Integer|The number of rows to aggregate before persisting. This number is the post-aggregation rows, so it is not equivalent to the number of input events, but the number of aggregated rows that those events result in. This is used to manage the required JVM heap size. Maximum heap memory usage for indexing scales with maxRowsInMemory * (2 + maxPendingPersists).|no (default == 75000)|
+|`maxRowsInMemory`|Integer|The number of rows to aggregate before persisting. This number is the post-aggregation rows, so it is not equivalent to the number of input events, but the number of aggregated rows that those events result in. This is used to manage the required JVM heap size. Maximum heap memory usage for indexing scales with maxRowsInMemory * (2 + maxPendingPersists).|no (default == 1000000)|
+|`maxBytesInMemory`|Long|The maximum number of bytes to keep in memory to aggregate before persisting. This is used to manage the required JVM heap size. |no (default == One-sixth of max JVM memory)|
```

Comment:
This should include a comment like maxRowsInMemory's that tells people that the actual max is going to be double (or more, if you set maxPendingPersists higher). It would also be nice to warn people that this is approximate. Maybe something like:

> The number of bytes to aggregate in-heap before persisting. This is based on a rough estimate of memory usage, not actual usage. The maximum heap memory usage for indexing is maxBytesInMemory * (2 + maxPendingPersists).